yasnippet.el

```diff
@@ -3467,17 +3467,23 @@ Move the overlays, or create them if they do not exit."
              ;; (overlay-put ov 'evaporate t)
              (overlay-put ov 'modification-hooks '(yas--on-protection-overlay-modification)))))))
 
-(defun yas--on-protection-overlay-modification (_overlay after? _beg _end &optional _length)
+(defvar yas--buffer-pre-modification)
```

Comment:
I see, I didn't think about `replace-regexp`. But anyway, the length check should be fine for what the protection overlay is trying to achieve.

Here's why: protection overlays are largely a hack designed to help prevent, but not completely avoid, commands or behaviours that destroy other parts (like mirrors) that the snippet depends on *while the snippet is being navigated*. Think things like `C-k`, for example. 

The somewhat heroic measures you are taking to strictly prevent that tiny overlay from being modified still leave the gate is wide open for other problems, i.e. other commands that do mischief somewhere else.

Anyway, if you really need to save contents, I'd suggest placing it in the overlay, as an overlay property. It's safer than a variable, although I can't think of situations where the variable would be particularly problematic (unless Emacs moves multithreaded someday). If for no better reason, the function is the only one making use of that information, which is a property of that particular overlay's history. So it should be stored there.

A better, more robust, and more useful approach, would be to place a modification hook in *those parts that need to be protected*, or, perhaps, disable the ones that have been destroyed, so the snippet could keep functioning with a limp. Not sure if this last one is a good idea though.

I'll take this in consideration in the `snippet-engine` rewrite...